### PR TITLE
Component test to expose workload with custom domain

### PR DESCRIPTION
**Pull request by cnvergence**
**Description**

Changes proposed in this pull request:

- Implement go test testing custom domain to expose workload
- Refactor code structure
- Improve GVR functions

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/15376

### CODE REVIEW

**Review by dariusztutaj — APPROVED** [review]
/meow

File: tests/components/api-gateway/features/e2e/custom_domain.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Feature: Exposing endpoint with a custom domain
+  Scenario: Register new custom domain on the cluster
+    Given there is an "google-credentials" DNS cloud credentials secret in "default" namespace
```
Comment (IwonaLanger) [inline tests/components/api-gateway/features/e2e/custom_domain.feature:3]:
```suggestion
    Given there is a "google-credentials" DNS cloud credentials secret in "default" namespace
```

File: tests/components/api-gateway/features/e2e/custom_domain.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    Then there is an "istio-ingressgateway" service in "istio-system" namespace
+    And create custom domain resources
+    And ensure that DNS record is ready
+
+  Scenario: Calling an unsecured API endpoint with custom domain
+    Given there is an unsecured endpoint
+    Then calling the endpoint without a token should result in status between 200 and 299
+    And calling the endpoint with any token should result in status between 200 and 299
+ 
+  Scenario: Calling an secured API with OAuth2
```
Comment (IwonaLanger) [inline tests/components/api-gateway/features/e2e/custom_domain.feature:13]:
```suggestion
  Scenario: Calling a secured API with OAuth2
```

File: tests/components/api-gateway/features/e2e/custom_domain.feature [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    Given endpoint is secured with OAuth2
+    Then calling the endpoint without a token should result in status beetween 400 and 403
+    And calling the endpoint with a invalid token should result in status beetween 400 and 403
```
Comment (IwonaLanger) [inline tests/components/api-gateway/features/e2e/custom_domain.feature:16]:
```suggestion
    And calling the endpoint with an invalid token should result in status beetween 400 and 403
```

**Review by dariusztutaj — APPROVED** [review]
/meow

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**